### DISCUSSION TITLE
Expressions should have a conditional operator or function

### DISCUSSION THREAD

**Discussion by enumag** [community/community#75928] [Actions] [answered]
### Select Topic Area

Product Feedback

### Body

[Expressions](https://docs.github.com/en/actions/learn-github-actions/expressions) should have a way to write a condition. This could either be a ternary operator `<condition> ? <true value> : <false value>` or a function `if(<condition>, <true-value>, <false-value>)`.

Original issue: https://github.com/actions/runner/issues/409

**Comment by axel-h** [comment]
In addition, the solution should support non-trivial use cases, Which breaks down into:
- handling multiple values for one expression
- an if-cascade

Since expressions are supposed to be reasonable simple, there is no need for a sophisticated solution. But some syntactical sugar would be nice.

**Comment by piotrekkr** [comment]
There seems to be a way because there is "ternary like" operator in Github actions https://docs.github.com/en/actions/learn-github-actions/expressions#example . One thing to watch for is that "true value" (one after `&&`) need to be "truthy" to work as expected

```
${{ github.event_name == 'release' && github.ref || inputs.release_ref }}
```

**Comment by axel-h** [comment]
This is well known, but considered a poor solution due to the side effects. A real ternary operator is needed.

**Reply by piotrekkr** [reply]
Okay. Then probably title/description need to be updated to emphasize on being a real ternary and not this "kind of" ternary that already is there.

**Comment by davetapley** [comment]
Following [the docs](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#example) I just wrote this (`legacy` is a bool):

```
${{ (matrix.legacy && '') || '.new' }}.example.com
```

Can you spot the problem? I didn't at first:

`''` is [**falsy**](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#literals), and so it falls through to `'.new'` _regardless_ of the value of `legacy`, causing `upload-artifact` to fail with:

```
Error: Failed to CreateArtifact: Received non-retryable error: Failed request: (409) Conflict: an artifact with this name already exists on the workflow run
```

Very confusing at first, and a 'real' ternary if would have avoided it.

**Reply by TrongBYM** [reply]
Hello, was wondering if you found a workaround for this specific case?

**Reply by davetapley** [reply]
@TrongBYM yeah I just inverted the conditional:

```
${{ (!matrix.legacy && '.new') || '' }}.example.com
```

Not pretty but it works.

**Comment by salilsub** [comment, marked as answer]
Hi everyone, we just [announced](https://github.blog/changelog/2026-01-29-github-actions-smarter-editing-clearer-debugging-and-a-new-case-function/#write-more-expressive-expressions-with-a-case-function) our new [`case` function](https://docs.github.com/en/actions/reference/workflows-and-actions/expressions#case) which is adds conditionals for expressions. Let us know if you have any feedback!
